pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -314,11 +314,45 @@ func (c *Cluster) resolveNode(nodeName, deviceClass string) *cephv1.Node {
 	if rookNode == nil {
 		return nil
 	}
-	rookNode.Resources = k8sutil.MergeResourceRequirements(rookNode.Resources, cephv1.GetOSDResources(c.spec.Resources, deviceClass))
 
+	rookNode.Resources = c.getOSDResources(rookNode, rookNode.ResourceClassName, deviceClass)
 	return rookNode
 }
 
+func (c *Cluster) resourceRequirementFromResourceClass(resourceClassName string) (corev1.ResourceRequirements, error) {
+	for _, i := range c.ValidStorage.ResourceClasses {
+		if i.Name == resourceClassName {
+			return i.Resources, nil
+		}
+	}
+	return corev1.ResourceRequirements{}, errors.Errorf("failed to find resourceClassName %s", resourceClassName)
+}
+
+func resolveDevice(devices []cephv1.Device, deviceClass string) string {
+	var deviceResourceClass = ""
+	for _, i := range devices {
```

Comment:
Why are we iterating over the devices? It seems like this is being called for a node, in which case we should just look for the resourceClassName at the node level instead of searching all its devices.

Reply:
The case when 'resourceClassName' is not set at the device level.
So applying the node level resource limits.

Reply:
If the `case when 'resourceClassName' is not set at the device level.`
we can also point towards the older field `Resources` which was set, instead of adding separate field ResourceClassName at Node-level.

What to do?

Reply:
Let's add a resourceClassName at the node level, and will have higher priority than the node resources. The resourceClassName is just a simple way to reference the resources that are defined in a single place, instead of duplicating the full resource spec everywhere.